engine/src/main/java/org/destinationsol/game/asteroid/FarAsteroid.java

```diff
@@ -80,13 +75,12 @@ private static Vector2 adjustDesiredPosition(SolGame game, Vector2 desiredPositi
                     else {
                         distanceVector.scl((size + .5f) / distance);
                     }
-                    newPosition = fromPosition.cpy().add(distanceVector);
-                    SolMath.free(SolMath.distVec(fromPosition, newPosition));
+                    position = fromPosition.cpy().add(distanceVector);
```

Comment:
You are discarding old position, and replacing it with a new vector that contains the same value as fromPosition + distanceVector. I think it's better to eliminate the new vector creation bit, and simply do
```position.set(fromPosition)
position.add(distanceVector)```
(Note that this prevents an object creation, which is huge.